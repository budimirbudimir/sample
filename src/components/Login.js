import React from 'react'

const Login = ({ loginError, submit, reset }) => {
	let email
	let pw

	const onSubmit = e => {
		e.preventDefault()

		if (email && pw) submit(email.value, pw.value)
	}

	const onReset = e => {
		e.preventDefault()

		if (email) reset(email.value)
	}

	return (
		<div className="Auth">
			{loginError && <p className="App-alert App-error">{loginError}</p>}

			<h3>Login</h3>

			<form onSubmit={onSubmit}>
				<div className="Auth-email_container">
					<input type="text" ref={el => (email = el)} placeholder="Email" />
				</div>

				<div className="Auth-password_container">
					<input type="password" placeholder="Password" ref={el => (pw = el)} />
				</div>

				<p>
					<a className="resetPassword" onClick={onReset}>
						Forgot Password?
					</a>
				</p>

				<button type="submit">Login</button>
			</form>
		</div>
	)
}

export default Login
