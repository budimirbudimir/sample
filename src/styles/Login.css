.Login {
	padding: 15px 0;
}

.Login h3 {
	margin: 35px auto;
	font-weight: 700;
}

.Login-email_container {
	margin: 0 auto 15px;
}

.Login-password_container {
	margin: 0 auto 15px;
}

.Login input,
.Login button {
	padding: 5px 15px;
	min-width: 300px;
	max-width: 100%;
	border-radius: 4px;
}
.Login input {
	border: 1px solid #ccc;
}
.Login button {
	width: 300px;
	background: #f4425c;
	color: #fff;
	border: none;
	letter-spacing: 1px;
	cursor: pointer;
	transition: all 0.3s;
}
.Login button:hover {
	opacity: 0.5;
}
.Login a.resetPassword {
	color: #f4425c;
	cursor: pointer;
	transition: all 0.3s;
}
.Login a.resetPassword:hover {
	color: #f4425c;
	text-decoration: underline;
}
