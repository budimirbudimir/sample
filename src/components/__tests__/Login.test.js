import React from 'react'
import { shallow } from 'enzyme'

import Login from '../Login'

const loginError = ''

describe('Login', () => {
	let wrapper

	beforeEach(() => {
		wrapper = shallow(<Login loginError={loginError} />)
	})

	it('matches snapshot', () => {
		expect(wrapper).toMatchSnapshot()
	})
})
