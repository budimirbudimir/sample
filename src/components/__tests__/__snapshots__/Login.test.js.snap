// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login matches snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Login
    loginError=""
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        "",
        <h3>
          Login
</h3>,
        <form
          onSubmit={[Function]}
>
          <div
                    className="Auth-email_container"
          >
                    <input
                              placeholder="Email"
                    />
          </div>
          <div
                    className="Auth-password_container"
          >
                    <input
                              placeholder="Password"
                              type="password"
                    />
          </div>
          <p>
                    <a
                              className="resetPassword"
                              onClick={[Function]}
                    >
                              Forgot Password?
                    </a>
          </p>
          <button
                    type="submit"
          >
                    Login
          </button>
</form>,
      ],
      "className": "Auth",
    },
    "ref": null,
    "rendered": Array [
      "",
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Login",
        },
        "ref": null,
        "rendered": "Login",
        "type": "h3",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="Auth-email_container"
>
              <input
                            placeholder="Email"
              />
</div>,
            <div
              className="Auth-password_container"
>
              <input
                            placeholder="Password"
                            type="password"
              />
</div>,
            <p>
              <a
                            className="resetPassword"
                            onClick={[Function]}
              >
                            Forgot Password?
              </a>
</p>,
            <button
              type="submit"
>
              Login
</button>,
          ],
          "onSubmit": [Function],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <input
                placeholder="Email"
/>,
              "className": "Auth-email_container",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "placeholder": "Email",
              },
              "ref": [Function],
              "rendered": null,
              "type": "input",
            },
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <input
                placeholder="Password"
                type="password"
/>,
              "className": "Auth-password_container",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "placeholder": "Password",
                "type": "password",
              },
              "ref": [Function],
              "rendered": null,
              "type": "input",
            },
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <a
                className="resetPassword"
                onClick={[Function]}
>
                Forgot Password?
</a>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Forgot Password?",
                "className": "resetPassword",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": "Forgot Password?",
              "type": "a",
            },
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Login",
              "type": "submit",
            },
            "ref": null,
            "rendered": "Login",
            "type": "button",
          },
        ],
        "type": "form",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          "",
          <h3>
            Login
</h3>,
          <form
            onSubmit={[Function]}
>
            <div
                        className="Auth-email_container"
            >
                        <input
                                    placeholder="Email"
                        />
            </div>
            <div
                        className="Auth-password_container"
            >
                        <input
                                    placeholder="Password"
                                    type="password"
                        />
            </div>
            <p>
                        <a
                                    className="resetPassword"
                                    onClick={[Function]}
                        >
                                    Forgot Password?
                        </a>
            </p>
            <button
                        type="submit"
            >
                        Login
            </button>
</form>,
        ],
        "className": "Auth",
      },
      "ref": null,
      "rendered": Array [
        "",
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Login",
          },
          "ref": null,
          "rendered": "Login",
          "type": "h3",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="Auth-email_container"
>
                <input
                                placeholder="Email"
                />
</div>,
              <div
                className="Auth-password_container"
>
                <input
                                placeholder="Password"
                                type="password"
                />
</div>,
              <p>
                <a
                                className="resetPassword"
                                onClick={[Function]}
                >
                                Forgot Password?
                </a>
</p>,
              <button
                type="submit"
>
                Login
</button>,
            ],
            "onSubmit": [Function],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <input
                  placeholder="Email"
/>,
                "className": "Auth-email_container",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "placeholder": "Email",
                },
                "ref": [Function],
                "rendered": null,
                "type": "input",
              },
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <input
                  placeholder="Password"
                  type="password"
/>,
                "className": "Auth-password_container",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "placeholder": "Password",
                  "type": "password",
                },
                "ref": [Function],
                "rendered": null,
                "type": "input",
              },
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <a
                  className="resetPassword"
                  onClick={[Function]}
>
                  Forgot Password?
</a>,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Forgot Password?",
                  "className": "resetPassword",
                  "onClick": [Function],
                },
                "ref": null,
                "rendered": "Forgot Password?",
                "type": "a",
              },
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Login",
                "type": "submit",
              },
              "ref": null,
              "rendered": "Login",
              "type": "button",
            },
          ],
          "type": "form",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
